BaragonService/src/main/java/com/hubspot/baragon/service/elb/ApplicationLoadBalancer.java

```diff
@@ -637,24 +637,29 @@ private void deregisterRemovableTargets(TrafficSource trafficSource,
     Collection<TargetDescription> removableTargets = listRemovableTargets(trafficSource, baragonGroup, targets, agents);
     LOG.info("removableTargets.size()={}", removableTargets.size());
 
+    Map<TargetDescription, TargetHealthDescription> targetToHealthDescriptionMap = getTargetToHealthDescriptionMap(targetGroup);
+
     for (TargetDescription removableTarget : removableTargets) {
       LOG.info("Processing removableTarget={}", removableTarget);
       try {
         if (configuration.isPresent()
             && !configuration.get().isRemoveLastHealthyEnabled()
-            && isLastHealthyInstance(removableTarget, targetGroup)) {
+            && isLastHealthyInstance(removableTarget, targetToHealthDescriptionMap)) {
           LOG.info("Will not de-register target {} because it is last healthy instance in {}", removableTarget, targetGroup);
```

Comment:
TBH i'm curious how this helps for large LB groups that _need_ to have multiple instances in them?

Should we have some configurable minimum here?

Reply:
So my understanding is that this is to prevent catastrophe like the one on Friday where there is nothing in the group serving requests, but accepts that during a ZK outage we might have very under-provisioned target groups.

Let's see, I can make this configurable in theory quite easily